save-demo/src/main/kotlin/com/saveourtool/save/demo/controller/DependencyController.kt

```diff
@@ -0,0 +1,118 @@
+     * @param projectName saveourtool project name
+     * @param version version that the file should be marked with
+     * @return [Flux] of [FileDto]s present in storage
+     */
+    @GetMapping("/{organizationName}/{projectName}/list-file")
+    fun listFiles(
+        @PathVariable organizationName: String,
+        @PathVariable projectName: String,
+        @RequestParam version: String,
+    ): Flux<FileDto> = demoService.findBySaveourtoolProjectOrNotFound(organizationName, projectName) {
+        "Could not find demo for $organizationName/$projectName."
+    }
+        .flatMapMany {
+            dependencyStorage.list(it, version)
+        }
+        .map { dependency ->
+            FileDto(
+                ProjectCoordinates(dependency.demo.organizationName, dependency.demo.projectName),
+                dependency.fileName,
+                LocalDateTime.now().toKotlinLocalDateTime(),
+            )
+        }
+
+    /**
+     * @param organizationName saveourtool organization name
+     * @param projectName saveourtool project name
+     * @param version version of a file to be deleted
+     * @param fileName name of a file to be deleted
+     * @return [Mono] of [Unit]
+     */
+    @DeleteMapping("/{organizationName}/{projectName}/delete-file")
+    fun deleteFile(
+        @PathVariable organizationName: String,
+        @PathVariable projectName: String,
+        @RequestParam version: String,
+        @RequestParam fileName: String,
+    ): Mono<Unit> = demoService.findBySaveourtoolProjectOrNotFound(organizationName, projectName) {
+        "Could not find demo for $organizationName/$projectName."
+    }
+        .flatMap { dependencyStorage.delete(it, version, fileName) }
+
+    /**
+     * @param organizationName saveourtool organization name
+     * @param projectName saveourtool project name
+     * @param version version to attach files to
+     * @param fileDtos list of [FileDto] containing information required for file download
+     * @return [Mono] of [StringResponse] with response
+     */
+    @PostMapping("/{organizationName}/{projectName}/upload")
+    fun uploadFiles(
+        @PathVariable organizationName: String,
+        @PathVariable projectName: String,
+        @RequestParam version: String,
+        @RequestBody fileDtos: List<FileDto>,
+    ): Mono<StringResponse> = demoService.findBySaveourtoolProjectOrNotFound(organizationName, projectName) {
+        "Could not find demo for $organizationName/$projectName."
+    }
+        .flatMapIterable { demo ->
+            fileDtos.map { Dependency(demo, version, it.name, it.requiredId()) }
+        }
+        .filterWhen {
+            dependencyStorage.doesExist(it).map(Boolean::not)
+        }
+        .collectList()
+        .map { dependencies ->
+            downloadToolService.downloadToStorage(dependencies).let { size ->
+                StringResponse("Successfully saved $size files to demo storage.", HttpStatus.OK)
```

Comment:
maybe add `StringResponse` when all `dependencies` already in storage? (all entities are filtered out by `dependencyStorage.doesExist(it).map(Boolean::not)`)

Reply:
Also will add in next PR as this one is already merged

Reply:
https://github.com/saveourtool/save-cloud/pull/1852